plugins/org.python.pydev.debug/src_console/org/python/pydev/debug/newconsole/env/PydevIProcessFactory.java

```diff
@@ -127,6 +131,56 @@ public PydevConsoleLaunchInfo createInteractiveLaunch() throws UserCanceledExcep
         if (activeEditor instanceof PyEdit) {
             edit = (PyEdit) activeEditor;
         }
+        String interpreterRep = InteractiveConsolePrefs.getInterpreterRepresentation();
+        if (!"none".equals(interpreterRep) || !interpreterRep.isEmpty()) {
```

Comment:
I don't think you need this... you can just set the UI elements (radio boxes) selections based on what was saved on the settings (when the UI is created) and then if the setting is valid just don't `open` the dialog (make `okPressed` public and call it directly instead of `open` and proceed with what was already implemented).

i.e.: there's no need to replicate the related logic -- also, there are `NullPointerExceptions` which are possible in this reimplementation (when there are multiple `interpreterInfos` configured, `interpreterInfo` may end up being `null`, but it's still accessed afterwards).